### ISSUE TITLE
The compose fails if any updates are kicked out due to failing test gating

### ISSUE THREAD

**Issue by bowlofeggs** [fedora-infra/bodhi#2368]
We started enforcing test gating in the composer in #2346. Today some of the composes failed after kicking some updates out of the compose for failing the test gating check, with this message:

Message
-------
[2018-05-09 14:35:37][bodhi.server   ERROR]
```python
Unable to determine requested tag for libnfs-1.11.0-1.fc27.
```

Process Details
---------------
- host:     bodhi-backend01.phx2.fedoraproject.org
- PID:      57207
- name:     fedmsg-hub
- command:  /usr/bin/python2 /usr/bin/fedmsg-hub
- msg_id:   

Callstack that lead to the logging statement
--------------------------------------------
```python
  File "/usr/lib64/python2.7/threading.py", line 777 in __bootstrap
    self.__bootstrap_inner()
  File "/usr/lib64/python2.7/threading.py", line 804 in __bootstrap_inner
    self.run()
  File "/usr/lib/python2.7/site-packages/bodhi/server/consumers/masher.py", line 373 in run
    self.work()
  File "/usr/lib/python2.7/site-packages/bodhi/server/consumers/masher.py", line 437 in work
    self.determine_and_perform_tag_actions()
  File "/usr/lib/python2.7/site-packages/bodhi/server/consumers/masher.py", line 69 in wrapper
    retval = method(self, *args, **kwargs)
  File "/usr/lib/python2.7/site-packages/bodhi/server/consumers/masher.py", line 589 in determine_and_perform_tag_actions
    self._determine_tag_actions()
  File "/usr/lib/python2.7/site-packages/bodhi/server/consumers/masher.py", line 621 in _determine_tag_actions
    move_tags.append((from_tag, update.requested_tag,
  File "/usr/lib/python2.7/site-packages/bodhi/server/models.py", line 3556 in requested_tag
    'Unable to determine requested tag for %s.' % self.title)
```

The compose later halted with this message:

Message
-------
[2018-05-09 14:35:39][fedmsg.consumers   ERROR]
```python
Exception in ComposerThread(f26-updates)
```

Process Details
---------------
- host:     bodhi-backend01.phx2.fedoraproject.org
- PID:      57207
- name:     fedmsg-hub
- command:  /usr/bin/python2 /usr/bin/fedmsg-hub
- msg_id:   

Callstack that lead to the logging statement
--------------------------------------------
```python
  File "/usr/lib64/python2.7/threading.py", line 777 in __bootstrap
    self.__bootstrap_inner()
  File "/usr/lib64/python2.7/threading.py", line 804 in __bootstrap_inner
    self.run()
  File "/usr/lib/python2.7/site-packages/bodhi/server/consumers/masher.py", line 373 in run
    self.work()
  File "/usr/lib/python2.7/site-packages/bodhi/server/consumers/masher.py", line 473 in work
    self.log.exception('Exception in ComposerThread(%s)' % self.id)
```
```
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/bodhi/server/consumers/masher.py", line 437, in work
    self.determine_and_perform_tag_actions()
  File "/usr/lib/python2.7/site-packages/bodhi/server/consumers/masher.py", line 69, in wrapper
    retval = method(self, *args, **kwargs)
  File "/usr/lib/python2.7/site-packages/bodhi/server/consumers/masher.py", line 590, in determine_and_perform_tag_actions
    self._perform_tag_actions()
  File "/usr/lib/python2.7/site-packages/bodhi/server/consumers/masher.py", line 652, in _perform_tag_actions
    failed_tasks = buildsys.wait_for_tasks([task[0] for task in results],
KeyError: 0
```

I believe this will happen relatively infrequently in the future, but could still happen if Greenwave changes its decision between the time the packager pushes an update to stable and the time the compose runs. Thus, we should fix it.

**Comment by bowlofeggs** [comment]
I am raising the priority on this as the severity is worse than I had realized when I wrote the ticket yesterday. I learned today that this traceback happened after several builds had their tags altered in Koji. When this traceback happened, those builds remained re-tagged in Koji but Bodhi did not record which ones had been altered. The next day when the compose was resumed, Bodhi ejected many updates from the compose, for example [this one](https://bodhi.fedoraproject.org/updates/FEDORA-2018-9dc3a15a3a):

> nodejs-6.14.2-1.fc26 ejected from the push because u"Cannot find relevant tag for nodejs-6.14.2-1.fc26. None of ['f26-updates', 'f26-updates-pending'] are in [u'f22-updates-testing', u'dist-6E-epel-testing', u'f21-updates-testing', u'f25-updates-testing', u'f24-updates-testing', u'epel7-testing', u'f27-modular-updates-testing', u'dist-5E-epel-testing', u'f23-updates-testing', u'f26-updates-testing', u'f28-updates-testing', u'f27-updates-testing', u'f28-modular-updates-testing']."

As of this morning, nodejs was tagged with f26-updates, but it was still considered "testing" by Bodhi.

**Comment by bowlofeggs** [comment]
I am working to fix the builds that have gotten into this state here:

https://pagure.io/fedora-infrastructure/issue/6925

**Comment by bowlofeggs** [comment]
The patch for this issue will be included in [```3.8.0```](https://bodhi.stg.fedoraproject.org/docs/user/release_notes.html#v3-8-0), which is currently deployed to [staging](https://bodhi.stg.fedoraproject.org).
